From c739dac1e224f6835e6a4ebdd9ad0568783bab4f Mon Sep 17 00:00:00 2001
From: stack <stack@apache.org>
Date: Tue, 22 Jul 2014 10:21:37 -0700
Subject: [PATCH] HBASE-11559 Add dumping of DATA block usage to the BlockCache
 JSON report (Michael Stack)

Amending-Author: Andrew Purtell <apurtell@apache.org>
---
 .../apache/hadoop/hbase/tmpl/regionserver/BlockCacheTmpl.jamon | 10 ++++++++++
 .../hadoop/hbase/tmpl/regionserver/BlockCacheViewTmpl.jamon    |  2 +-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/BlockCacheTmpl.jamon b/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/BlockCacheTmpl.jamon
index 5e94279..a042318 100644
--- a/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/BlockCacheTmpl.jamon
+++ b/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/BlockCacheTmpl.jamon
@@ -325,6 +325,16 @@ are combined counts. Request count is sum of hits and misses.</p>
         <td>Size of DATA Blocks</td>
     </tr>
 </%if>
+    <tr>
+        <td>Evicted</td>
+        <td><% String.format("%,d", bc.getStats().getEvictedCount()) %></td>
+        <td>The total number of blocks evicted</td>
+    </tr>
+    <tr>
+        <td>Evictions</td>
+        <td><% String.format("%,d", bc.getStats().getEvictionCount()) %></td>
+        <td>The total number of times an eviction has occurred</td>
+    </tr>
 <%doc>Can't do age of block in cache when slab cache</%doc>
 <%if !slabCache %>
     <tr>
diff --git a/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/BlockCacheViewTmpl.jamon b/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/BlockCacheViewTmpl.jamon
index 50da12a..6fc24b2 100644
--- a/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/BlockCacheViewTmpl.jamon
+++ b/hbase-server/src/main/jamon/org/apache/hadoop/hbase/tmpl/regionserver/BlockCacheViewTmpl.jamon
@@ -55,7 +55,7 @@ com.yammer.metrics.stats.Snapshot;
   }
   CachedBlocksByFile cbsbf = BlockCacheUtil.getLoadedCachedBlocksByFile(conf, bc);
 </%java>
-<%if bcv.equals("file") %><& bc_by_file; cbsbf = cbsbf; &><%else><% BlockCacheUtil.toJSON(bc) %></%if>
+<%if bcv.equals("file") %><& bc_by_file; cbsbf = cbsbf; &><%else>{<% BlockCacheUtil.toJSON(bc) %>, <% cbsbf %> }</%if>
 <%java>
 cbsbf = null;
 </%java>
-- 
1.9.1

